src/creator/core.rs

```diff
@@ -0,0 +1,200 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use tracing::{error, warn};
+
+/// Represents any type of state that can be retrieved and encoded
+pub trait State: Debug + Send + Sync + Clone + Default {}
+
+/// Generic default state implementation for primitive types
+/// This can be used by any state type that wraps a primitive with a default value
+#[derive(Debug, Clone, PartialEq, Eq, Default, Hash)]
+pub struct DefaultState<T: Default + Clone + Send + Sync + std::fmt::Debug>(pub T);
+
+impl<T: Default + Clone + Send + Sync + std::fmt::Debug> State for DefaultState<T> {}
+
+/// Represents task data that can be encoded into a payload
+pub trait TaskData: Debug + Send + Sync {
+    /// Encode this task data into a payload
+    fn encode_into_payload(self, encoded_state: Vec<u8>) -> Vec<u8>;
+}
+
+/// A provider that can retrieve the current state
+#[async_trait]
+pub trait StateProvider: Send + Sync {
```

Comment:
I can see this is downstream implemented in the counter but i still don't think we should make an assumption or introduce a state component for the creator, i suggest leaving that implemented in the counter